src/LanguageServer/ProtocolUnitTests/DataTips/DataTipRangeHandlerTests.cs

```diff
@@ -0,0 +1,61 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Roslyn.Test.Utilities;
+using Xunit;
+using Xunit.Abstractions;
+using LSP = Roslyn.LanguageServer.Protocol;
+
+namespace Microsoft.CodeAnalysis.LanguageServer.UnitTests.DataTips;
+
+public class DataTipRangeHandlerTests(ITestOutputHelper testOutputHelper)
+    : AbstractLanguageServerProtocolTests(testOutputHelper)
+{
+    private static async Task<LSP.VSInternalDataTip?> RunAsync(TestLspServer testLspServer, LSP.Location caret)
+    {
+        return await testLspServer.ExecuteRequestAsync<LSP.TextDocumentPositionParams, LSP.VSInternalDataTip?>(
+            LSP.VSInternalMethods.TextDocumentDataTipRangeName,
+            new LSP.TextDocumentPositionParams()
+            {
+                TextDocument = new LSP.TextDocumentIdentifier { Uri = caret.Uri },
+                Position = caret.Range.Start,
+            },
+            CancellationToken.None);
+    }
+
+    [Theory, CombinatorialData]
+    public async Task SimpleStatement(bool mutatingLspWorkspace)
+    {
+        var markup = """
+            using System.Linq;
+
+            int[] args;
+            var v = args.{|caret:|}Select(a => a.ToString()).Where(a => a.Length >= 0);
+            """;
+        await using var testLspServer = await CreateTestLspServerAsync(markup, mutatingLspWorkspace);
+
+        var caret = testLspServer.GetLocations("caret").Single();
+
+        var result = await RunAsync(testLspServer, caret);
+        Assert.NotNull(result);
+
+        AssertJsonEquals(new LSP.VSInternalDataTip
```

Comment:
should we consider making `VSInternalDataTip` a record type so we could just equals these?

Reply:
sure!